### PR TITLE
unsized constant assign

### PR DESCRIPTION
**Pull request by alaindargelas**

### CODE REVIEW

File: src/DesignCompile/Elaboration_test.cpp [hunk 1]
```diff
@@ -474,12 +475,14 @@ endmodule
       const std::string& name = cassign->Lhs()->VpiName();
       if (name == "p1") {
         // Val is 1, but it has a signed typespec (Meaning negative bin number)
-        EXPECT_EQ(val, 1);
+        EXPECT_EQ(val, 3);
```
Comment (hzeller) [inline src/DesignCompile/Elaboration_test.cpp:478]: why did this value change ?

Comment (alaindargelas) [reply src/DesignCompile/Elaboration_test.cpp:478]: Previous value was incorrect as unsigned where partially supported so far

Comment (alaindargelas) [reply src/DesignCompile/Elaboration_test.cpp:478]: Now they get expended based on LHS size

File: src/DesignCompile/CompileHelper.cpp [hunk 2]
```diff
@@ -2948,6 +2948,26 @@ bool CompileHelper::compileParameterDeclaration(
   return true;
 }
 
+char flip(char c) { return (c == '0') ? '1' : '0'; }
+
+std::string twosComplement(std::string bin) {
```
Comment (hzeller) [inline src/DesignCompile/CompileHelper.cpp:2953]: Always use std::string_view as parameter unless absolutely not possible.

**Pull request merged**